File: CS/10_StampsAndWatermarks/ImageAndPageNumber.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using Spire.Pdf.AutomaticFields;

namespace ImageAndPageNumber
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //create a PDF document
            PdfDocument doc = new PdfDocument();
            doc.PageSettings.Size = PdfPageSize.A4;

            //reset the default margins to 0
            doc.PageSettings.Margins = new PdfMargins(0);

            //create a PdfMargins object, the parameters indicate the page margins you want to set
            PdfMargins margins = new PdfMargins(50, 50, 50, 50);

            //get page size
            SizeF pageSize = doc.PageSettings.Size;

            //create a header template with content and apply it to page template
            doc.Template.Top = CreateHeaderTemplate(doc, margins, pageSize);

            //create a footer template with content and apply it to page template
            doc.Template.Bottom = CreateFooterTemplate(doc, margins, pageSize);

            //apply blank templates to other parts of page template
            doc.Template.Left = new PdfPageTemplateElement(margins.Left, doc.PageSettings.Size.Height);
            doc.Template.Right = new PdfPageTemplateElement(margins.Right, doc.PageSettings.Size.Height);

           //Create one page
            PdfPageBase page = doc.Pages.Add();            

            //Draw the text
            page.Canvas.DrawString("Hello, World!",
                                   new PdfFont(PdfFontFamily.Helvetica, 30f),
                                   new PdfSolidBrush(Color.Black),
                                   10, 10);

            //save the file
            string output = "ImageandPageNumberinHeaderFootersection_out.pdf";
            doc.SaveToFile(output,FileFormat.PDF);

            PDFDocumentViewer(output);
        }
        private PdfPageTemplateElement CreateHeaderTemplate(PdfDocument doc, PdfMargins margins, SizeF pageSize)
        {
            //create a PdfPageTemplateElement object as header space
            PdfPageTemplateElement headerSpace = new PdfPageTemplateElement(pageSize.Width, margins.Top);
            headerSpace.Foreground = false;

            //declare two float variables
            float x = margins.Left;
            float y = 0;

            //draw image in header space 
            PdfImage headerImage = PdfImage.FromFile("../../../../../../../Data/E-iceblueLogo.png");
            float width = headerImage.Width / 2;
            float height = headerImage.Height / 2;
            headerSpace.Graphics.DrawImage(headerImage, x, margins.Top - height - 5, width, height);

            //draw line in header space
            PdfPen pen = new PdfPen(PdfBrushes.LightGray, 1);
            headerSpace.Graphics.DrawLine(pen, x, y + margins.Top - 2, pageSize.Width - x, y + margins.Top - 2);

            //return headerSpace
            return headerSpace;
        }

        private PdfPageTemplateElement CreateFooterTemplate(PdfDocument doc, PdfMargins margins, SizeF pageSize)
        {
            // Create a PdfPageTemplateElement object to serve as the footer space
            PdfPageTemplateElement footerSpace = new PdfPageTemplateElement(pageSize.Width, margins.Bottom);
            footerSpace.Foreground = false;

            // Declare two float variables for positioning
            float x = margins.Left;
            float y = 0;

            // Draw a line in the footer space using a gray pen
            PdfPen pen = new PdfPen(PdfBrushes.Gray, 1);
            footerSpace.Graphics.DrawLine(pen, x, y, pageSize.Width - x, y);

            // Draw text in the footer space
            y = y + 5;

            // Define the font for the text
            PdfTrueTypeFont font = new PdfTrueTypeFont(new Font("Arial", 10f), true);
            //////////////////Use the following code for netstandard dlls/////////////////////////
            /*
             PdfTrueTypeFont font = new PdfTrueTypeFont("Arial", 10f, PdfFontStyle.Regular, true);
            */

            // Create a dynamic field for the page number
            PdfPageNumberField number = new PdfPageNumberField();

            // Create a dynamic field for the page count
            PdfPageCountField count = new PdfPageCountField();

            // Create a composite field that combines the page number and page count fields
            PdfCompositeField compositeField = new PdfCompositeField(font, PdfBrushes.Black, "Page {0} of {1}", number, count);

            // Set the string format for the composite field (left-aligned, top vertical alignment)
            compositeField.StringFormat = new PdfStringFormat(PdfTextAlignment.Left, PdfVerticalAlignment.Top);

            // Measure the size of the composite field to determine its bounds
            SizeF size = font.MeasureString(compositeField.Text);
            compositeField.Bounds = new RectangleF(x, y, size.Width, size.Height);

            // Draw the composite field on the footer space
            compositeField.Draw(footerSpace.Graphics);

            // Return the footer space element
            return footerSpace;
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB/10_StampsAndWatermarks/ImageAndPageNumber.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics
Imports Spire.Pdf.AutomaticFields

Namespace ImageAndPageNumber
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new instance of PdfDocument
			Dim doc As New PdfDocument()

			' Set the page size to A4
			doc.PageSettings.Size = PdfPageSize.A4

			' Set the page margins to zero
			doc.PageSettings.Margins = New PdfMargins(0)

			' Define the margins for the header and footer
			Dim margins As New PdfMargins(50, 50, 50, 50)

			' Get the page size from the document's page settings
			Dim pageSize As SizeF = doc.PageSettings.Size

			' Create a header template for the document using the specified margins and page size
			doc.Template.Top = CreateHeaderTemplate(doc, margins, pageSize)

			' Create a footer template for the document using the specified margins and page size
			doc.Template.Bottom = CreateFooterTemplate(doc, margins, pageSize)

			' Create a left template for the document with the specified left margin and full page height
			doc.Template.Left = New PdfPageTemplateElement(margins.Left, doc.PageSettings.Size.Height)

			' Create a right template for the document with the specified right margin and full page height
			doc.Template.Right = New PdfPageTemplateElement(margins.Right, doc.PageSettings.Size.Height)

			' Add a new page to the document
			Dim page As PdfPageBase = doc.Pages.Add()

			' Draw a string on the page as "Hello, World!" using the specified font, brush, and position
			page.Canvas.DrawString("Hello, World!", New PdfFont(PdfFontFamily.Helvetica, 30.0F), New PdfSolidBrush(Color.Black), 10, 10)

			' Specify the output file path
			Dim output As String = "ImageandPageNumberinHeaderFootersection_out.pdf"

			' Save the document to the output file in PDF format
			doc.SaveToFile(output, FileFormat.PDF)

			' Close the document
			doc.Close()

			' Launch the file
			PDFDocumentViewer(output)
		End Sub
		Private Function CreateHeaderTemplate(ByVal doc As PdfDocument, ByVal margins As PdfMargins, ByVal pageSize As SizeF) As PdfPageTemplateElement
			' Create a new PdfPageTemplateElement with the specified width and top margins
			Dim headerSpace As New PdfPageTemplateElement(pageSize.Width, margins.Top)
			headerSpace.Foreground = False

			' Set the initial coordinates for drawing elements
			Dim x As Single = margins.Left
			Dim y As Single = 0

			' Load the header image from file
			Dim headerImage As PdfImage = PdfImage.FromFile("../../../../../../../Data/E-iceblueLogo.png")

			' Calculate the width and height of the header image
			Dim width As Single = headerImage.Width \ 2
			Dim height As Single = headerImage.Height \ 2

			' Draw the header image at the specified position with the calculated size
			headerSpace.Graphics.DrawImage(headerImage, x, margins.Top - height - 5, width, height)

			' Create a pen for drawing lines with light gray color and thickness of 1
			Dim pen As New PdfPen(PdfBrushes.LightGray, 1)

			' Draw a line across the page using the pen
			headerSpace.Graphics.DrawLine(pen, x, y + margins.Top - 2, pageSize.Width - x, y + margins.Top - 2)

			' Return the header template
			Return headerSpace
		End Function

		Private Function CreateFooterTemplate(ByVal doc As PdfDocument, ByVal margins As PdfMargins, ByVal pageSize As SizeF) As PdfPageTemplateElement
			' Create a new PdfPageTemplateElement with the specified width and bottom margins
			Dim footerSpace As New PdfPageTemplateElement(pageSize.Width, margins.Bottom)
			footerSpace.Foreground = False

			' Set the initial coordinates for drawing elements
			Dim x As Single = margins.Left
			Dim y As Single = 0

			' Create a pen for drawing lines with gray color and thickness of 1
			Dim pen As New PdfPen(PdfBrushes.Gray, 1)

			' Draw a line across the page using the pen
			footerSpace.Graphics.DrawLine(pen, x, y, pageSize.Width - x, y)

			' Update the y-coordinate for positioning the next element
			y = y + 5

			' Create a font object with Arial font and size 10
			Dim font As New PdfTrueTypeFont(New Font("Arial", 10.0F), True)
			' =============================================================================
			' Use the following code for netstandard dlls
			' =============================================================================
			'Dim font As New PdfTrueTypeFont("Arial", 10.0F, FontStyle.Regular, True)
			' =============================================================================

			' Create page number and page count fields
			Dim number As New PdfPageNumberField()
			Dim count As New PdfPageCountField()

			' Create a composite field with the specified font, color, format, and content
			Dim compositeField As New PdfCompositeField(font, PdfBrushes.Black, "Page {0} of {1}", number, count)
			compositeField.StringFormat = New PdfStringFormat(PdfTextAlignment.Left, PdfVerticalAlignment.Top)

			' Measure the size of the composite field to calculate its bounds
			Dim size As SizeF = font.MeasureString(compositeField.Text)
			compositeField.Bounds = New RectangleF(x, y, size.Width, size.Height)

			' Draw the composite field on the footer template
			compositeField.Draw(footerSpace.Graphics)

			' Return the footer template
			Return footerSpace
		End Function
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
